pkg/envoy/xds/ack.go

```diff
@@ -194,10 +196,12 @@ func (m *AckingResourceMutatorWrapper) Delete(typeURL string, resourceName strin
 	version, _, revert := m.mutator.Delete(typeURL, resourceName, true)
 
 	if c == nil {
-		log.WithFields(logrus.Fields{
-			logfields.XDSTypeURL:      typeURL,
-			logfields.XDSResourceName: resourceName,
-		}).Fatal("no completion given to delete xDS resource")
+		// No completion in upsert, revert also without completion
+		return func(_ *completion.Completion) {
```

Comment:
I think it would be simpler and more consistent to just use with the revert func below. Even if this mutation didn't have a completion, it doesn't mean that you should ignore any completion for reverting it.
Just remove this whole `if` block here, and enclose the stuff below into an `if c != nil`:
```
if c != nil {
    if _, found := m.pendingCompletions[c]; found {
        ...
    }
    m.addDeleteCompletion(typeURL, version, nodeIDs, c)
}
```